### ISSUE TITLE
USB Decoder does not function correctly.

### ISSUE THREAD

**Issue by Codysseus** [azonenberg/scopehal#361]
I wrote a simple test case for the USB decoder that the offset of a given packet plus the length did not exceed the offset of the next packet. However, when I tried this on my own data I found that this did not hold true.
[Here](https://web.engr.oregonstate.edu/~hollidac/usbcsv.cpp) is a link to the modified `usbcsv` example with the test, and [here](web.engr.oregonstate.edu/~hollidac/init.7z) is a link to the 7z that contains the csv in question. Be careful as the CSV is almost 2GB.

**Comment by azonenberg** [comment]
Just checking whether the fixes I mentioned on IRC solved this?

**Comment by Codysseus** [comment]
Hey Andrew! Thank you for following up with this. At the moment I do not have the bandwidth to debug, but I will get back to you on Tuesday (Dec 08 2020) with what I have tried. I hope you had a nice weekend!

**Comment by Codysseus** [comment]
So one thing RE: my sample. Channel 0-Analog is D-, and Channel 1-Analog is D+. I was able to load a Low Speed sample into glscopeclient and it seemed to label the sample correctly. I flipped the channels and set it to Low Speed, now packets are emitted, but there is still an overlap issue. Here is a picture.

![image](https://user-images.githubusercontent.com/9455568/102271142-cf304200-3ed3-11eb-9795-6e2ce44282f5.png)
